nslsii/__init__.py

```diff
@@ -71,8 +71,13 @@ def configure_base(user_ns, broker_name, *,
     # Set up a RunEngine and use metadata backed by a sqlite file.
     from bluesky import RunEngine
     from bluesky.utils import get_history
-    RE = RunEngine(get_history())
-    ns['RE'] = RE
+    # if RunEngine already defined grab it
+    # useful when users make their own custom RunEngine
+    if 'RE' in user_ns:
+        RE = user_ns['RE']
+    else:
+        RE = RunEngine(get_history())
+        ns['RE'] = RE
```

Comment:
I would recommend this should be used in both cases.

Reply:
Either way works. We have user_ns.update(ns) in the end.

Reply:
OK

Reply:
@licode is right that we update `user_ns`either way, but there is a small difference. We return `ns` at the end to provide a way for the user to know which keys in the namespace we altered. I like the current implementation in this PR because it only includes `'RE'` in `ns` if we altered it.